File: CSharpFundamentals/CSharpOOPbasics/PolymorphismLab/Shapes/Rectangle.cs
namespace Shapes
{
    public class Rectangle : Shape
    {
        private double width;
        private double height;

        public double Height
        {
            get { return height; }
            set { height = value; }
        }

        public double Width
        {
            get { return width; }
            set { width = value; }
        }

        public Rectangle(double width, double height)
        {
            Width = width;
            Height = height;
        }

        public override double CalculateArea()
        {
            return Width * Height;
        }

        public override double CalculatePerimeter()
        {
            return (Width + Height) * 2;
        }

        public override string Draw()
        {
            return base.Draw() + "Rectangle";
        }
    }
}

File: JavaFundamentals/JavaOOP/Polymorphism-Lab/src/shape/Rectangle.java
package shape;

public class Rectangle extends Shape {
    private Double height;
    private Double width;

    public Rectangle(Double height, Double width) {
        this.setHeight(height);
        this.setWidth(width);
        this.calculateArea();
        this.calculatePerimeter();
    }

    public final Double getHeight() {
        return this.height;
    }

    private void setHeight(Double height) {
        this.height = height;
    }

    public final Double getWidth() {
        return this.width;
    }

    private void setWidth(Double width) {
        this.width = width;
    }

    @Override
    public void calculatePerimeter() {
        this.setPerimeter((this.height + this.width) * 2);
    }

    @Override
    public void calculateArea() {
        this.setArea(this.height * this.width);
    }
}
